prow/cmd/deck/main.go

```diff
@@ -1172,18 +1188,28 @@ func handleRerun(prowJobClient prowv1.ProwJobInterface, createProwJob bool) http
 		// On Prow instances that require auth even for viewing Deck this is okayish, because the Prowjob UUID
 		// is hard to guess
 		// Ref: https://github.com/kubernetes/test-infra/pull/12827#issuecomment-502850414
-		if createProwJob {
-			if r.Method != http.MethodPost {
```

Comment:
So previously this would allow only a POST if we were creating a new prowjob. Now, we do nothing if it isn't POST but we don't error. Why?

Reply:
If it isn't POST, we do the same thing `/rerun` does now (display the config YAML). This preserves the functionality for `kubectl create -f "https://prow.k8s.io/rerun?prowjob=blahblah"` and isn't necessarily an error. If someone tries to directly trigger a job but doesn't use a POST request, they'll just see the YAML which I think seems reasonable.